basic_system/src/system_implementation/system/da_commitment_generator/blob_commitment_generator/commitment_and_proof_advice.rs

```diff
@@ -0,0 +1,63 @@
+pub const BLOB_COMMITMENT_AND_PROOF_QUERY_ID: u32 =
+    zk_ee::oracle::query_ids::ADVICE_SUBSPACE_MASK | 0x20;
+
+#[cfg(target_arch = "riscv32")]
+pub fn blob_commitment_and_proof_advice(
+    data: &[u8],
+    oracle: &mut impl zk_ee::oracle::IOOracle,
+) -> [u8; 96] {
+    let mut it = oracle
+        .raw_query(
+            BLOB_COMMITMENT_AND_PROOF_QUERY_ID,
+            &(data.as_ptr() as usize as u32, data.len() as u32),
+        )
+        .unwrap();
+
+    let mut buffer = [0u8; 96];
+    unsafe {
+        let buffer_u32_ptr: *mut u32 = buffer.as_mut_ptr().cast::<[u32; 24]>().cast();
+        for i in 0..24 {
+            buffer_u32_ptr.add(i).write(it.next().unwrap() as u32)
+        }
+    }
+    assert!(it.next().is_none());
+    buffer
+}
+
+#[cfg(not(target_arch = "riscv32"))]
+pub fn blob_commitment_and_proof_advice(
+    data: &[u8],
+    _oracle: &mut impl zk_ee::oracle::IOOracle,
+) -> [u8; 96] {
+    use crypto::MiniDigest;
+
+    let sidecar_builder: alloy_consensus::SidecarBuilder<alloy_consensus::SimpleCoder> =
+        alloy_consensus::SidecarBuilder::from_slice(data);
+    // TODO: at this step we compute also kzg proof, which is not needed in fact
```

Comment:
Should we avoid it now? Or as a follow-up?

Reply:
Oh, wait, this TODO shouldn't be here indeed, I have to update this function generally, thanks for flagging